OneSignal/OneSignal-iOS-SDK
# [JWT] Get In-App Messages from Server

# Description
## One Line Summary
Update fetch IAMs to use the new endpoint, support JWT, and include the alias.

## Details

### Motivation
Fetch IAMs

### Scope
Fetch IAMs

### Future Work
* The user and subscription model's relationship should be improved. Should the subscription model own a reference to its identity model that owns it?
* The retrying fetch IAMs can be improved and less hacky.

# Testing
## Unit testing
No changes

## Manual testing
New install with JWT on and off, misc scenarios tested
- Works under normal flows with connection.
- With JWT on, receive a 401 from fetch IAM. Then update the jwt token, and see the fetch that is immediately retried.
- With JWT off, login to a new user with network turned off, and start a new session. Fetch IAM is blocked by `null` onesignal ID. Turn data on and the user is created. The fetch IAM is immediately retried.

# Affected code checklist

# Checklist
## Overview
     - If it is hard to explain how any codes changes are related to each other then it most likely needs to be more than one PR

## Testing

## Final pass
      - Simplify with less code, followed by splitting up code into well named functions and variables, followed by adding comments to the code.
      - WIP (Work In Progress) is ok, but explain what is still in progress and what you would like feedback on. Start the PR title with "WIP" to indicate this.


- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/OneSignal/OneSignal-iOS-SDK/1476)
